legal-api/src/legal_api/core/filing.py

```diff
@@ -61,17 +62,44 @@ def id(self) -> str:  # pylint: disable=invalid-name; defining the std ID
                 self._id = self._storage.id
         return self._id
 
+    @property
+    def filing_type(self) -> str:
+        """Property containing the filing type."""
+        if not self._filing_type and self._storage:
+            self._filing_type = self._storage.filing_type
+        return self._filing_type
+
     @property
     def raw(self) -> Optional[Dict]:
         """Return the raw, submitted and unprocessed version on the filing."""
         if not self._raw:
-            return {}
+            if self._storage:
+                self._raw = self._storage.json
+            else:
+                return {}
         return self._raw
 
     @property
     def json(self) -> Optional[Dict]:
         """Return a dict representing the filing json."""
-        return self._raw
+        _json = {}
+        if self._storage:
+            if self._storage.status == Filing.Status.COMPLETED.value:
+                _json = VersionedBusinessDetailsService. \
+                    get_revision(self.filing_type, self.id, self._storage.business_id)
```

Comment:
Since you are passing the filing id, and then retrieving the filing, do we need to pass in the type?

Reply:
That's true. Will change it.